astro/src/content/quickstarts/quickstart-go-api.mdx

```diff
@@ -0,0 +1,377 @@
+  "email": "teller@example.com",
+  "email_verified": true,
+  "applicationId": "e9fdb985-9173-4e01-9d73-ac2d60d1dc8e",
+  "roles": [
+    "teller"
+  ],
+  "auth_time": 1689285985,
+  "tid": "d7d09513-a3f5-401c-9685-34ab6c552453"
+}
+```
+
+You need to validate the token and make sure the user has the appropriate role to access the API. You can do so by adding the following code:
+
+<RemoteCode url={frontmatter.coderoot + "/complete-application/main.go"} tags="authorization"
+  lang="go"/>
+
+There is a lot of code in this function. Let's break it down:
+- First you validate some claims. Notice the `checkAudience` that makes sure the token is intended for this application and `checkIss` to make sure the issuer is as expected.
+- `SetPublicKey`, which is a function we will create later, retrieves the public key for the JWT.
+- After the key has been set, the `golang-jwt/jwt` will validate the signature and expiration date.
+- Once the token is validated, you will need to take a look at the roles.
+- The code will look to see which page is being called and then create an array for the roles that are valid for that request in `validRoles`.
+     - The `containsRole` function will be created as well and that creates an array with the intersection of roles that are allowed and roles that are required. If the length of that intersection is greater than 1 then the user has the proper role.
+- Once the roles have been validated, then the endpoint is served.
+
+Now, you will add the code to set the public key. This function will accept the key Id from the token and call the authorization server to get the public key with that key Id and put it in the proper format to be used.
+
+<Aside type="note">
+Different tokens can be signed with different keys. In this example there is only one key used. If the key is already set, the application will not attempt to set it again. In a production environment you could store this key on the server or set it once on initialization. If your api serves several applications, you will need to get the correct public key for each application if the signing keys are different between applications.
+</Aside>
+
+<RemoteCode url={frontmatter.coderoot + "/complete-application/main.go"} tags="set-public-key"
+  lang="go"/>
+
+Then add the code to compare the required roles to the user's roles based on the token.
+
+<RemoteCode url={frontmatter.coderoot + "/complete-application/main.go"} tags="contains-role"
+  lang="go"/>
+
+Finally we will add a helper function that returns the name of the function being called by the endpoint.
+
+<RemoteCode url={frontmatter.coderoot + "/complete-application/main.go"} tags="helper"
+  lang="go"/>
+
+That is all the code for the sample application. Save the `main.go` file.
+
+## Run the API
+
+Start the API resource server by running:
+
+In a command shell window, navigate to the root directory of your API.
+
+```shell
+cd your-application
+go get
+go run main.go
+```
```

Comment:
I get this error message when I step through the instructions:

```
dan@mooreds your-application % go run main.go
# command-line-arguments
./main.go:53:15: undefined: ChangeResponse
./main.go:94:28: undefined: DenominationCounter
```

I am using 1.21.

I don't see this when I go into `complete-application`. There it works fine.